### DISCUSSION TITLE
How do I enable Discussions on a repo?

### DISCUSSION THREAD

**Discussion by TonyValenti** [community/community#24] [Discussions] [answered]
How do I enable Discussions on a repo?

**Comment by becca** [comment]
We're still in a limited public beta. As soon as we launch as a public beta you will have the ability to enable Discussions on a public repository. 

In the meantime, you can always request a specific public repository be added to the beta list.

**Reply by TonyValenti** [reply]
Can you enable it on https://github.com/abpframework/abp ?

**Reply by TonyValenti** [reply]
And what is the timeline on a general release?

**Reply by becca** [reply]
I just flagged your repository into the beta - you can now toggle Discussions on under your repo's `/settings`! 
No exact ETA on a general release but we're working on it 🚀 Be on the lookout

**Reply by gabek** [reply]
Hi Becca, would you be able to add https://github.com/owncast/owncast to the Discussions beta, as well? :)

**Reply by yajo** [reply]
Please add https://github.com/copier-org/copier, https://github.com/Tecnativa/doodba and https://github.com/Tecnativa/doodba-copier-template

**Reply by becca** [reply]
🎸 got it, flagged these in so you can now toggle the feature on for the aforementioned repositories

**Reply by kevwan** [reply]
Please add https://github.com/tal-tech/go-zero, thanks!

**Reply by berstend** [reply]
@becca apologies for hijacking this thread, if you'd be able to flag https://github.com/berstend/puppeteer-extra for the beta program that would be splendid :-) thanks a lot!

**Reply by ghost** [reply]
I don't know if you are still taking requests, but https://github.com/BenSova/Patched-Sur might like discussions for whenever a new version of macOS comes out or general discussions about requested/coming/how to use features. Who knows what may happen with it, but I feel it might be a good idea.

**Reply by Daniellop1** [reply]
Hello, could I also add my repository? https://github.com/Dglobalcode/dglobal/tree/main

**Reply by zeuchste** [reply]
@becca can you please also enable our repository for the beta of discussions? https://github.com/nebulastream/nebulastream/

**Reply by ghost** [reply]
@zeuchste It's a public beta now, go into settings then you can enable it.

**Reply by zeuchste** [reply]
@BenSova unfortunately, I cannot find it under settings, maybe you can point me to the right option?

**Reply by ghost** [reply]
@zeuchste Sorry for the late reply, if you haven't already figured it out, it's in the same spot with the options for whether or not you want to have a wiki or issues.

**Reply by bryan7238** [reply]
Could you please enable discussions for the public repo arithmetric/aws-lambda-ses-forwarder
Thank you!!

**Comment by eggebraa** [comment]
@becca It looks like this may be the place to request access to testing the Discussions beta feature? If so, our team is very interested in trying it out. We have a number of repositories, but the one we'd like to try it on first would be: https://github.com/Alvearie/quality-measure-and-cohort-service

**Reply by becca** [reply]
👋 this space is for feedback but I went ahead and flagged this repository into the beta.

**Comment by Juliaria08** [comment]
Can you please add [this](https://github.com/Julian08-Code/octo-tests/) repo under the beta program?

**Reply by Juliaria08** [reply]
Or beter the org because i want to test all of this on test repos

**Reply by becca** [reply]
We can only flag individual repositories in at this time.

**Reply by PranamBhat** [reply]
Thank you for the update!

**Comment by elizabetdev** [comment]
Hi @becca 👋 ,

Today I was searching for a solution to better organize the discussions in our repo [Elastic UI](https://github.com/elastic/eui/issues?q=is%3Aopen+is%3Aissue+label%3Adiscussion). We have a lot of issues and a few of them are just **discussions** where we label with `discussion`:

<img width="1242" alt="Screenshot 2020-11-04 at 16 43 39" src="https://user-images.githubusercontent.com/2750668/98140270-24346d80-1ebd-11eb-9561-3b645efb7b5c.png">
  
It seems that this beta feature would be really helpful for us. Is it possible to enable this beta feature into our repo? https://github.com/elastic/eui

**Reply by becca** [reply]
👋 hey there! thankfully we also have a way to convert all open issues with any label - if you navigate to the labels page you'll see the `convert issues` button to convert all those _discussions_. 

I flagged your repository into the beta so you can test things out 🚀

**Reply by elizabetdev** [reply]
Thanks @becca 🎉

**Reply by dbarrosop** [reply]
Is this still true? Can't seem to find the option to `convert issues` based on labels

**Comment by srid** [comment]
Could I request to enable Discussions Beta for [neuron](https://github.com/srid/neuron)? Cheers.

**Reply by becca** [reply]
yes, you can! just flagged that repository in so you can toggle Discussions on now from the settings tab.

**Comment by vtolstov** [comment]
can you enable discussions on github.com/unistack-org/micro ? Thanks!

**Comment by jiangz222** [comment]
Can you enable discussion on https://github.com/qiniu/qmgo?
Thanks!

**Comment by ppwwyyxx** [comment]
Can you enable discussion on https://github.com/facebookresearch/detectron2/? It's a very active project. Thanks!

**Comment by kevwan** [comment]
Could I request to enable Discussions Beta for https://github.com/tal-tech/go-zero?
Thanks!

**Comment by gotom22** [comment]
@becca could you clarify whether there is a distinction between Discusssions at the organisation level vs at the repo level? 

I see that repos (like this one) already have some basic sorting features for discussions, whereas the Discussions feature I see in our private Organization (WHO-HEAT) is more rudimentary. 
- will these eventually be the same?
- could I get the more advanced repo version activated for WHO-HEAT/HEAT_extras? (even though it's private)
- Do you have a rough timeline for launching discussion features for public and private repos/orgs?

(Background:
We are setting up an internal discussion group (advisory group) which would be ok at the org or repo level. But ideally we will not have to change the setup once started. We would like to get going in December. We are debating between Issues in a new/clean repo, Discussions as available at the org level, or, preferably the most advanced version of Discussions with some sorting/labeling features etc. (it would be ok if these only became available within the next weeks or months).

**Reply by becca** [reply]
Yes, great question! This is Discussions at the REPOSITORY level only right now. 

The Upvoting feature is still in a limited beta and we hope to 🚢 it to everyone shortly, which includes all the sorting functionality improvements as well).

We do not yet have an official ETA for launching Discussions for private repositories but please be on the lookout for more in our [public roadmap](github.com/github/roadmap).

**Reply by dpieski** [reply]
The public roadmap link is not working for me. Is that link correct?

**Reply by manudona92** [reply]
Hi Becca, 

Might you enable Discussions panel to https://github.com/manudona92/MapantESP  ?

We are developing a big mapping project that covers the whole territory of Spain. It would be a very handy tool for our contributors. 

Cheers

**Reply by ghost** [reply]
@manudona92 It's in public beta now, just go into settings and scroll down a little bit then you'll see it!

**Reply by cecruzmo** [reply]
> The public roadmap link is not working for me. Is that link correct?

@dpieski The link is malformed. It seems like the correct link is https://github.com/github/roadmap

**Comment by triclambert** [comment]
Hi @becca, we are planning to open source our `opstrace/opstrace` repo on 12/1.  Is there any chance we can get in the queue to have discussions added then?  We want to use them as our primary communication mechanism for our OSS project.  Launching without discussions could be...  a challenge.  :)  What do you think?

**Comment by rstoenescu** [comment]
Hi @becca , can you enable GH discussions for https://github.com/quasarframework/quasar pls? We plan on switching to it from our own forum. Thank you!

**Comment by dbarrosop** [comment]
Hi @becca, would you mind enabling discussions for https://github.com/nornir-automation/nornir too, please? Thanks for the awesome work you do.

**Comment by TonyValenti** [comment]
Hi @becca - Would you please enable discussions on https://github.com/emgucv/emgucv ?

**Comment by zeuchste** [comment]
Hi @becca can you maybe please also enable discussions for us? https://github.com/nebulastream/nebulastream

**Comment by Daniellop1** [comment]
Hi, discussions is now available for everyone

**Comment by becca** [comment, marked as answer]
👋 We are officially launched as a beta for all public repositories to opt into. You can view more information here in our [changelog post](https://github.blog/changelog/2020-12-08-github-discussions-now-available-as-a-public-beta/) as well as the [blog post](https://github.blog/2020-12-08-new-from-universe-2020-dark-mode-github-sponsors-for-companies-and-more/).

For some additional details regarding next steps regarding private repositories and specifics around upvoting etc please reference this above thread https://github.com/github/feedback/discussions/24#discussioncomment-179134

**Reply by zeuchste** [reply]
@becca ok but do you know when it will be available also for private repositories?

**Reply by becca** [reply]
It's mentioned in the thread I linked in this comment. No ETA at the moment and you can watch the public roadmap to know when it's coming 🚀

**Reply by karliss** [reply]
Maybe it should have a "beta" label next to the checkbox in repository options just like "Allow auto-merge" has. When I saw "Discussions" appearing  in options for public repositories without "beta" indicator (while other features in the same page has it) I initially thought that discussions is finished and production ready.

**Reply by ThomasAunvik** [reply]
So i got a request to add Discussions to one of my repositories, but whenever i tried to press "Setup Discussions" i get this following error.
![image](https://user-images.githubusercontent.com/9968151/110598005-8e1d9600-8181-11eb-92a6-27b98ce8deb5.png)
![image](https://user-images.githubusercontent.com/9968151/110598122-b0afaf00-8181-11eb-9375-5d5f41623609.png)

**Comment by ekovegeance** [comment]
@teakopp

**Comment by ekovegeance** [comment]
Github discussion is still in the testing stage

**Comment by relrelway** [comment]
@becca Github Discussions - **https://github.com/sleepyproject/Starlight-Jet/**

**Comment by Colonizor48** [comment]
can you add [this](https://github.com/Colonizor48/stellaris-scrapped-origins-reimagined-mod) to it? The repo might be small but i intend for it to be bigger.

**Reply by dpieski** [reply]
You have to enable discussions yourself in settings. See the selected answer for more

**Comment by rebelvc** [comment]
Hi Github, can you please add this to the beta program? Would love to try it out.

**Reply by ghost** [reply]
Just open Settings and look for Discussions. It's in public beta now, so you can try it yourself!

**Reply by rebelvc** [reply]
Thanks but i don't see it here.

<img width="1051" alt="Screen Shot 2021-01-28 at 8 29 43 AM" src="https://user-images.githubusercontent.com/1904841/106145005-046eb780-6143-11eb-855a-d193e8fa48b4.png">

**Reply by dpieski** [reply]
It is only available in Public repos, not private repos.

**Reply by ag-monk** [reply]
Can you please add this repo under beta? https://github.com/ag-monk/action

**Comment by ag-monk** [comment]
Hi, 
Can you please add this repo under beta? https://github.com/ag-monk/action

**Comment by ljcremer** [comment]
Hi, Can you please enable it on the following repo:
https://github.com/Trooptravel/tt-ui-web

**Comment by luisfelipesdn12** [comment]
Settings > Options > Features > Discussions > Enjoy :)

**Comment by fdevans** [comment]
In January there was a comment that this was only available for Public repositories.  Is that still the case?  I have a "internal" repository that we are preparing for "public" but wanted to get all the Discussions parts configured prior to releasing.

**Reply by fdevans** [reply]
Currently if I try to enable it I get an error message and there is no Start Discussion button.

**Comment by leonwanghui** [comment]
Could I request to enable Discussions Beta for https://github.com/tinyms-ai/specification and https://github.com/tinyms-ai/awesome-rusted-ai?
Thanks a lot!

**Comment by pkishino** [comment]
could you please enabled discussions and issues beta for https://github.com/haugene/docker-transmission-openvpn please?

**Comment by shah-aditya** [comment]
Is this available for GitHub Enterprise or is planned in the future?

**Comment by ghost** [comment]
@becca could you please flag this repo to enable discussions beta- https://github.com/jayachandra27/FunctionalBundle

Thanks.

**Comment by jdnichollsc** [comment]


**Reply by jdnichollsc** [reply]
Nevermind, it's working now! ❤️

**Comment by AJESH-CR** [comment]


**Reply by laymonage** [reply]
It's already available to all users. You just need to activate it in [your repository's settings](https://github.com/AJESH-CR/solo-learn/settings).

![image](https://user-images.githubusercontent.com/6379424/124426606-7901ee00-dd94-11eb-8164-306e97516445.png)

**Comment by parashardhapola** [comment]
Hi,

Is it possible to have the Discussion feature activated for this repo?: [Scarf](https://github.com/parashardhapola/scarf)

Thanks!

**Reply by laymonage** [reply]
Yes, it's already available to all users. You just need to activate it in [your repository's settings](https://github.com/parashardhapola/scarf/settings).

![image](https://user-images.githubusercontent.com/6379424/124426606-7901ee00-dd94-11eb-8164-306e97516445.png)

**Comment by gerhardwagner** [comment]
Hi,

I have enabled Discussions on a enterprise internal repo
I can access discussions if i go to the url `<repo>/discussions` but it does not show the discussions tab on the repo view
am I doing anything wrong ?

Thanks in advance :-)

**Reply by laymonage** [reply]
Might be related to https://github.com/github/feedback/discussions/4897

**Comment by tecowong** [comment]
Sorry I am new to github. Where can I enable the discussion? I can found it in the setting. Can you please help to enable this
https://github.com/tecowong/Blynk_Mblock

**Reply by laymonage** [reply]
https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/enabling-or-disabling-github-discussions-for-a-repository

![Discussions](https://user-images.githubusercontent.com/6379424/124426606-7901ee00-dd94-11eb-8164-306e97516445.png)

**Reply by piiskop** [reply]
i don't see _Features_ at all in any of my repository.

**Comment by aleks-ivanov** [comment]
Is it possible to enable Discussions through the GitHub API, similar to `has_issues` and `has_wiki` ?

**Comment by schmurtzm** [comment]
Hi, Could you please enabled discussions and issues beta for https://github.com/schmurtzm/MrDiy-Audio-Notifier/  please?
And if possible to https://github.com/schmurtzm/RFLink32-For-Sonoff-RF-Bridge
Thank you.

**Comment by re-masashi** [comment]
Could you kindly add discussions to https://github.com/re-masashi/willpyre.
Thanks in advance.

**Reply by laymonage** [reply]
It's available for everyone: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/enabling-or-disabling-github-discussions-for-a-repository

**Reply by re-masashi** [reply]
Oh thanks. I am enabling it then.
